builtin/logical/ssh/path_sign.go

```diff
@@ -275,14 +274,20 @@ func (b *backend) calculateKeyId(data *framework.FieldData, req *logical.Request
 		return reqId, nil
 	}
 
-	keyHash := sha256.Sum256(pubKey.Marshal())
-	keyId := hex.EncodeToString(keyHash[:])
+	keyIdFormat := "vault-{{token_display_name}}-{{public_key_hash}}"
+	if req.DisplayName == "" {
+		keyIdFormat = "vault-{{public_key_hash}}"
+	}
 
-	if req.DisplayName != "" {
-		keyId = fmt.Sprintf("%s-%s", req.DisplayName, keyId)
+	if role.KeyIDFormat != "" {
+		keyIdFormat = role.KeyIDFormat
 	}
 
-	keyId = fmt.Sprintf("vault-%s", keyId)
+	keyId := substQuery(keyIdFormat, map[string]string{
+		"token_display_name": req.DisplayName,
```

Comment:
If req.DisplayName is an empty string, the return value should be the same. Here's an example: https://play.golang.org/p/um-WzsD8RE